### PR TITLE
Adding new Brion version with support for libsonata

### PR DESCRIPTION
**Pull request by NadirRoGue**

### CODE REVIEW

**Comment by ppodhajski** [comment]
retest this please

File: var/spack/repos/builtin/packages/brion/package.py [hunk 1]
```diff
@@ -30,6 +31,8 @@ class Brion(CMakePackage):
     depends_on('boost +shared', when='~python')
     depends_on('boost +shared +python', when='+python')
 
+    depends_on('libsonata', when='@3.1.0')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/brion/package.py]:
```suggestion
    depends_on('libsonata', when='@3.1.0:')
```
I assume this dependency will stay in the future. Then we should use an open range (:

File: var/spack/repos/builtin/packages/brion/package.py [hunk 2]
```diff
@@ -39,7 +42,12 @@ class Brion(CMakePackage):
     depends_on('mvdtool ~mpi')
 
     def cmake_args(self):
-        return ['-DDISABLE_SUBPROJECTS=ON']
+        args = ['-DDISABLE_SUBPROJECTS=ON']
+
+        if '+libsonata' in self.spec:
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/brion/package.py]: What is this supposed to do? Since you have no `libsonata` variant, it shouldn't get triggered.

**Review by matz-e — COMMENTED** [review]
It quit with `package blender not found`. Are you on the latest spack? Seems like maybe a rebase is in order…

**Comment by matz-e** [comment]
Please update the PR description to something like `brion: add sonata support` or similiar. Just `Brion sonata` is not all that meaningful when looking at commit logs.

**Comment by ppodhajski** [comment]
retest this please

**Review by tristan0x — COMMENTED** [review]
If you want to bump the brion module deployed on BB5 to 3.1.0, you have to update `packages/bbp-packages.yaml` as well. Also I guess 3.1.0 should be the one _prefered_.

File: var/spack/repos/builtin/packages/brion/package.py [hunk 3]
```diff
@@ -39,7 +42,12 @@ class Brion(CMakePackage):
     depends_on('mvdtool ~mpi')
 
     def cmake_args(self):
-        return ['-DDISABLE_SUBPROJECTS=ON']
+        args = ['-DDISABLE_SUBPROJECTS=ON']
+
+        if self.version >= Version('3.1.0'):
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/brion/package.py]:
```suggestion
        if self.spec.satisfies('@3.1.0:'):
```

**Review by pramodk — CHANGES_REQUESTED** [review]
and this is what Tristan pointed:

```
→ grep -r brion deploy/*
deploy/packages/bbp-packages.yaml:      - brion@3.0.0 +python ^boost@1.65.0
```

**Review by pramodk — APPROVED** [review]

**Comment by matz-e** [comment]
> If you want to bump the brion module deployed on BB5 to 3.1.0, you have to update `packages/bbp-packages.yaml` as well. Also I guess 3.1.0 should be the one _prefered_.

The newest version is always preferred.

**Pull request merged**